enclave/crypto/cert.c

```diff
@@ -738,16 +740,36 @@ oe_result_t oe_cert_chain_read_pem(
         OE_RAISE(OE_INVALID_PARAMETER);
 
     /* Must have pem_size-1 non-zero characters followed by zero-terminator */
-    if (strnlen((const char*)pem_data, pem_size) != pem_size - 1)
-        OE_RAISE(OE_INVALID_PARAMETER);
+    // Note: the above comment is incorrect. There is no trailing zero,
+    // but mbedtls_x509_crt_parse() requires a trailing zero
+    OE_TRACE_INFO(
+        "strlen()=%d vs pem_size=%d",
+        strnlen((const char*)pem_data, pem_size),
+        pem_size);
+
+    uint8_t* tmp_pem_data = (uint8_t*)pem_data;
+    size_t tmp_pem_size = pem_size;
+    if (strnlen((const char*)pem_data, pem_size) == pem_size)
+    {
+        tmp_pem_size = pem_size + 1;
+        if (!(tmp_pem_data = oe_calloc(1, tmp_pem_size)))
```

Comment:
Do we need to do a calloc of it is immediately followed by a memcpy? I think oe_malloc is fine

Reply:
Sure.